types/module/configurator.go

```diff
@@ -15,16 +19,35 @@ type Configurator interface {
 	// QueryServer returns a grpc.Server instance which allows registering services
 	// that will be exposed as gRPC services as well as ABCI query handlers.
 	QueryServer() grpc.Server
+
+	// RegisterMigration registers an in-place store migration for a module. The
+	// handler is a migration script to perform in-place migrations from version
+	// `fromVersion` to version `fromVersion+1`.
+	RegisterMigration(moduleName string, fromVersion uint64, handler func(store sdk.KVStore) error) error
+
+	// RunMigrations runs all in-place store migrations for a module from a
+	// given version to another version.
+	RunMigrations(ctx sdk.Context, moduleName string, fromVersion, toVersion uint64) error
```

Comment:
This shouldn't be on the public interface of Configurator because it's only called by the module manager.

Reply:
sounds good, however I needed to cast Configurator (the interface) to configurator (the struct) to access private store keys. 2d554ce